aioshelly/__init__.py

```diff
@@ -1,187 +1,30 @@
-
-# Firmware 1.8.0 release date
-MIN_FIRMWARE_DATE = 20200812
-FIRMWARE_PATTERN = re.compile(r"^(\d{8})")
-
-DEVICE_TIMEOUT_SEC = 10
-
-
-class ShellyError(Exception):
-    """Base class for aioshelly errors."""
-
-
-class AuthRequired(ShellyError):
-    """Raised during initialization if auth is required but not given."""
-
-
-class NotInitialized(ShellyError):
-    """Raised if device is not initialized."""
-
-
-class FirmwareUnsupported(ShellyError):
-    """Raised if device firmware version is unsupported."""
-
-
-@dataclass
-class ConnectionOptions:
-    """Shelly options for connection."""
-
-    ip_address: str
-    username: Optional[str] = None
-    password: Optional[str] = None
-    temperature_unit: str = "C"
-    auth: Optional[aiohttp.BasicAuth] = None
-
-    def __post_init__(self):
-        """Call after initialization."""
-        if self.username is not None:
-            if self.password is None:
-                raise ValueError("Supply both username and password")
-
-            object.__setattr__(
-                self, "auth", aiohttp.BasicAuth(self.username, self.password)
-            )
-
-
-async def get_info(aiohttp_session: aiohttp.ClientSession, ip_address):
-    """Get info from device through REST call."""
-    async with aiohttp_session.get(
-        f"http://{ip_address}/shelly", raise_for_status=True
-    ) as resp:
-        result = await resp.json()
-
-    if (
-        "fw" not in result
-        or not supported_firmware(result["fw"])
-        or result["type"] in ["SHSW-44", "SHSEN-1"]
-    ):
-        raise FirmwareUnsupported
-
-    return result
-
-
-def supported_firmware(ver_str: str):
-    """Return True if device firmware version is supported."""
-    match = FIRMWARE_PATTERN.search(ver_str)
-
-    if match is None:
-        return False
-
-    # We compare firmware release dates because Shelly version numbering is
-    # inconsistent, sometimes the word is used as the version number.
-    return int(match[0]) >= MIN_FIRMWARE_DATE
+from .common import ConnectionOptions
+from .exceptions import ShellyError
+from .rpc_device import RpcDevice
 
 
 class Device:
     """Shelly device reppresentation."""
 
-    def __init__(
-        self,
-        coap_context: COAP,
-        aiohttp_session: aiohttp.ClientSession,
-        options: ConnectionOptions,
-    ):
-        """Device init."""
-        self.coap_context = coap_context
-        self.aiohttp_session = aiohttp_session
-        self.options = options
-        self.coap_d = None
-        self.blocks = None
-        self.coap_s = None
-        self._settings = None
-        self.shelly = None
-        self._status = None
-        self._unsub_listening = coap_context.subscribe_updates(
-            options.ip_address, self._coap_message_received
-        )
-        self._update_listener = None
-        self._coap_response_events: dict = {}
-        self._initialized = False
-        self._initializing = False
-
     @classmethod
     async def create(
```

Comment:
This makes no sense to keep this on this device. Instead, I suggest we split the function and move it to `BlockDevice` and `RpcDevice`, also because the parameters are different.